presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/CanonicalizeExpressionRewriter.java

```diff
@@ -107,5 +111,29 @@ public Expression rewriteExtract(Extract node, Void context, ExpressionTreeRewri
 
             throw new UnsupportedOperationException("not yet implemented: " + node.getField());
         }
+
+        @Override
+        public Expression rewriteFunctionCall(FunctionCall node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+        {
+            if (!REWRITE_ENABLED_FUNCTIONS.contains(node.getName())) {
+                return null;
+            }
+            ImmutableList.Builder<Expression> rewriteArguments = ImmutableList.builder();
+            boolean nested = false;
+            for (Expression arg : node.getArguments()) {
+                if (arg instanceof FunctionCall && ((FunctionCall) arg).getName().equals(node.getName())) {
+                    nested = true;
+                    rewriteArguments.addAll(((FunctionCall) arg).getArguments());
+                }
+                else {
+                    rewriteArguments.add(arg);
+                }
+            }
+            if (nested) {
+                FunctionCall rewriteNode = new FunctionCall(node.getName(), rewriteArguments.build().asList());
+                return treeRewriter.rewrite(rewriteNode, context);
+            }
+            return null;
```

Comment:
Same as above `null` comment. Should we return `null` or the original expression?